src/dibit_vector.jl

```diff
@@ -0,0 +1,74 @@
+"""
+    DiBitVector <: AbstractVector{UInt8}
+
+A bitvector whose elements are two bits wide, allowing
+storage of integer values between 0 and 3. Optimized
+for performance and memory savings.
+"""
+
+const UINT_SZ = sizeof(UInt)   # number of bytes in a system int: 8 or 4
+const IDX = UINT_SZ ÷ 2 + 1    # 5 for 64, 3 for 32
+const OFFSET = UINT_SZ * 8 - 1 # 63 for 64, 31 for 32
+
+mutable struct DiBitVector <: AbstractVector{UInt8}
+    data::Vector{UInt}
+    len::UInt
+
+    function DiBitVector(n::Integer, v::Integer)
+        if !(Int(v) in 0:3)
+            throw(ArgumentError("v must be in 0:3"))
+        end
+        fv = (0x0000000000000000, 0x5555555555555555,
+        0xaaaaaaaaaaaaaaaa, 0xffffffffffffffff)[v + 1]
+        vec = Vector{UInt}(undef, cld(n, 32))
+        fill!(vec, fv)
```

Comment:
This can't work if `UInt == UInt32`, as e.g. `0x5555555555555555` can't be losslessly converted to `UInt32`. You could use `fv % UInt`, but I would personally just use `UInt64` everywhere for simplicity, even if this means it's less efficient on 32-bits machines. This is e.g. what `BitSet` and `BitArray` use.

Reply:
I guess the problem lies in how do define it here @sbromberger . Can you split it case-wise. Like one for `UInt32` and other for `UInt64`.

Reply:
OK, I think I did it.

Reply:
Actually, I like the idea of just making things UInt64.